VTXdigiDetailed/src/VTXdigitizerDetailed.cpp

```diff
@@ -0,0 +1,976 @@
+    hChargePerCluster->SetYTitle("Clusters");
+    hChargePerCluster->SetDirectory(0);
+
+    /// Drift due to magnetic field
+
+    hXDriftDueToMagField = new TH1D("hXDriftDueToMagField", "X Drift due to magnetic field", 100, -0.04, 0.04);
+    hXDriftDueToMagField->SetXTitle("X Drift due to magnetic field (mm)");
+    hXDriftDueToMagField->SetDirectory(0);
+
+    hYDriftDueToMagField = new TH1D("hYDriftDueToMagField", "Y Drift due to magnetic field", 100, -0.04, 0.04);
+    hYDriftDueToMagField->SetXTitle("Y Drift due to magnetic field (mm)");
+    hYDriftDueToMagField->SetDirectory(0);
+
+    // Cluster and fired pixels per layer
+
+    hClusterPerLayer  = new TH1D("hClusterPerLayer", "Cluster per Layer", 10, 0, 10); 
+    hClusterPerLayer->SetDirectory(0);
+
+    hActivePixelPerlayer = new TH1D("hActivePixelPerlayer", "Active Pixels per Layer", 10, 0, 10);
+    hActivePixelPerlayer->SetXTitle("Layer");
+    hActivePixelPerlayer->SetYTitle("Number of active pixels");
+    hActivePixelPerlayer->SetDirectory(0);
+
+  }
+  
+  // Initialize random services
+  m_randSvc = service("RndmGenSvc", false);
+  if (!m_randSvc) {
+    error() << "Couldn't get RndmGenSvc!" << endmsg;
+    return StatusCode::FAILURE;
+  }
+
+  // Initialize Gaussian random generator for time smearing
+  m_gauss_t_vec.resize(m_t_resolution.size());
+  for (size_t i = 0; i < m_t_resolution.size(); ++i) {  
+    if (m_gauss_t_vec[i].initialize(m_randSvc, Rndm::Gauss(0., m_t_resolution[i])).isFailure()) {
+      error() << "Couldn't initialize RndmGenSvc!" << endmsg;
+      return StatusCode::FAILURE;
+    }
+  }
+
+  // Initialize Gaussian random generator for threshold smearing
+  if (m_gauss_threshold.initialize(m_randSvc, Rndm::Gauss(m_Threshold, m_ThresholdSmearing)).isFailure()) {
+    error() << "Couldn't initialize Gaussian generator for threshold smearing!" << endmsg;
+    return StatusCode::FAILURE;
+  }
+
+  // Initialize and check if GeoSvc is valid
+  m_geoSvc = serviceLocator()->service("GeoSvc");
+  if (!m_geoSvc) {
+    error() << "Failed to retrieve GeoSvc" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  info() << "GeoSvc successfully retrieved" << endmsg;
+
+  // Get the readout name from "inputSimHits" if "readoutName" not set by the user
+  if (m_readoutName.value().empty()) { m_readoutName = inputLocations("inputSimHits")[0]; }
+
+  // Check if the readout exists
+  if (m_geoSvc->getDetector()->readouts().find(m_readoutName.value()) == m_geoSvc->getDetector()->readouts().end()) {
+    error() << "Readout '" << m_readoutName.value() << "' does not exist in the geometry! Please Provide a readout name through 'readoutName' property" << endmsg;
+    return StatusCode::FAILURE;  // Or handle it gracefully
+  }
+
+  //Set the cellID decoder
+  try {
+    m_decoder = std::make_unique<dd4hep::DDSegmentation::BitFieldCoder>(
+        m_geoSvc->getDetector()->readout(m_readoutName.value()).idSpec().fieldDescription()
```

Comment:
The field description is part of the metadata associated to the input hits collection, you don't need to get that from the geoSvc via the readout name.

Reply:
https://github.com/key4hep/k4GaudiPandora/blob/b2d086b14e41f860dba021da404ae914ce217d35/k4GaudiPandora/src/DDSimpleMuonDigi.cc#L98-L99
```c++
  const auto encodingString =
      k4FWCore::getParameter<std::string>(collName + "__" + edm4hep::labels::CellIDEncoding, this);
```